### PR TITLE
Feature/df 2194 autosave

### PR DESCRIPTION
**Pull request by ganqqwerty**

### CODE REVIEW

File: alv-portal-ui/src/app/competence-catalog/competence-sets/competence-set-detail-edit/competence-set-edit.component.ts [hunk 1]
```diff
@@ -103,20 +97,30 @@ export class CompetenceSetEditComponent extends CompetenceCatalogEditorAwareComp
     }
   }
 
+  public loadFullSet(): void {
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/competence-sets/competence-set-detail-edit/competence-set-edit.component.ts]: I don't think you need this method

File: alv-portal-ui/src/app/competence-catalog/shared/competence-set/competence-set.component.html [hunk 2]
```diff
@@ -61,22 +61,43 @@
...
-  <div [@expandCollapse]="isCollapsed ? 'closed' : 'open'"
-       [alvCollapse]="isCollapsed"
+  <div [@expandCollapse]="isInitiallyCollapsed ? 'closed' : 'open'"
+       [alvCollapse]="isInitiallyCollapsed"
        [id]="competenceSet.id"
        class="panel-content">
 
-    <alv-competence-items-collapse-panel (actionClick)="addCompetenceElement(subElementType)"
-                                         *ngFor="let subElementType of subElementTypes"
+    <alv-competence-items-collapse-panel (actionClick)="addCompetenceElement(elementType.KNOW_HOW_INDICATOR)"
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/shared/competence-set/competence-set.component.html]: how come you're not using the subElementType forloop anymore?

**Comment by emina-str** [comment]
- changing statuses doesn't trigger any action, it seems.
- the cancel button on edit is not renamed to something like "back to list" (as defined in the ticket). leaving it as "Cancel" could be misleading.

File: alv-portal-ui/src/app/competence-catalog/shared/competence-set/competence-set.component.html (hunk 2 above)
Comment (ganqqwerty) [reply alv-portal-ui/src/app/competence-catalog/shared/competence-set/competence-set.component.html]: I duplicate that fragment now

File: alv-portal-ui/src/app/competence-catalog/competence-sets/competence-set-detail-edit/competence-set-edit.component.ts (hunk 1 above)
Comment (ganqqwerty) [reply alv-portal-ui/src/app/competence-catalog/competence-sets/competence-set-detail-edit/competence-set-edit.component.ts]: thanks, will remove

File: alv-portal-ui/src/app/competence-catalog/competence-sets/competence-set-detail-edit/competence-set-edit.component.ts [hunk 3]
```diff
@@ -120,16 +119,24 @@ export class CompetenceSetEditComponent extends CompetenceCatalogEditorAwareComp
...
+    });
     this.notificationsService.success('portal.competence-catalog.competence-sets.added-success-notification');
   }
 
   private handleFailure(error: HttpErrorResponse): Observable<never> {
     return this.businessExceptionsHandlerService.handleError(error);
   }
 
-  private openBacklinkModal(competenceSetSearchResult: CompetenceSetSearchResultDto) {
+  private openBacklinkModal(competencecompetenceSet: CompetenceSetWithResolvedKnowHow) {
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/competence-sets/competence-set-detail-edit/competence-set-edit.component.ts]: double competence

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.ts [hunk 4]
```diff
@@ -79,6 +87,9 @@ export class ChFicheComponent extends CompetenceCatalogEditorAwareComponent impl
   @Input()
   isReadonly = false;
 
+  @Output()
+  ficheChange = new EventEmitter<ChFiche>();
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.ts]: shouldn't it be "chFiche"?

**Comment by emina-str** [comment]
just one more thing
image
when there is an error with changing statuses, you should reset back to before state.
For example, here when I'm in Freigeben and want to go to Publiziert, but there is an error (that's fine). It stays on Publiziert (not OK), and when I click again on In Bearbeitung, I get this error, even though it wasn't successfully saved as Published (notice also that Publishing options are also disabled):
image

**Pull request merged**